tests/telemetry/test_telemetry.py

```diff
@@ -221,27 +221,6 @@ def process_trace(self, trace):
     ]["message"]
 
 
-def test_app_started_error_unhandled_tracer_exception(test_agent_session, run_python_code_in_subprocess):
-    env = os.environ.copy()
-    env["DD_SPAN_SAMPLING_RULES"] = "invalid_rules"
-
```

Comment:
Maybe we can keep this test if we add:
```
env["DD_TESTING_RAISE"] = True
```
So `config._raise` is set to True? And have a TODO to add another test to ensure without that config, there is no exception raised?

Reply:
This change replaces all of the exceptions from `DD_SPAN_SAMPLING_RULES` with warning logs, so regardless of that variable's value there won't be an exception raised. Because the intention of this PR is to make the library never raise an unhandled exception, the premise of this test isn't really valid anymore, which is why I chose to delete it.